tests/robustness/snapmeta/snapmeta.go

```diff
@@ -0,0 +1,22 @@
+// Package snapmeta describes entities that can accept
+// arbitrary metadata and flush it to a persistent repository.
+package snapmeta
+
+import "github.com/kopia/kopia/tests/robustness/snap"
+
+// Store describes the ability to store and retrieve
+// a buffer of metadata, indexed by a string key.
+type Store interface {
+	Store(key string, val []byte) error
+	Load(key string) ([]byte, error)
+	Delete(key string)
+	GetKeys() []string
+}
+
+// DataPersister describes the ability to flush metadata
+// to, and load it again, from a repository.
+type DataPersister interface {
```

Comment:
nit: `Persister`? It'd be referenced as `snapmeta.Persister` from other packages.

Also, should this interface include `Store` as well?
Part of the question is when does it make sense to have a persister that is not (or is independent of) a store?

Reply:
`snapmeta.Persister` sounds good.

Yeah had to ponder this one for a bit. My original intent for these two interfaces was a segregation of responsibility: The Checker only needs to know the `Store` interface, and the `Engine` only needs to know how to load/flush metadata (never actually needs to deal with snapIDs, so doesn't need the `Store` interface).

If we frame it slightly differently though: the Engine needs something that knows how to persist data (`Persister` interface), but also something that implements the `Store` interface that the Engine can give to its Checker. That would justify the hierarchy of including `Store` in `Persister`.